src/tcgwars/logic/impl/gen4/PopSeries6.groovy

```diff
@@ -87,358 +92,89 @@ public enum PopSeries6 implements LogicCardInfo {
-            text "Search your deck for a basic Energy card, show it to your opponent, and put it into your hand. Shuffle your deck afterward."
-            energyCost C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-          move "Thunder Jolt", {
-            text "20 damage. Flip a coin. If tails, Pachirisu does 10 damage to itself."
-            energyCost L
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (DiamondPearl.PACHIRISU_35, this);
       case RAMPARDOS_5:
-        return evolution (this, from:"Cranidos", hp:HP120, type:FIGHTING, retreatCost:1) {
-          weakness G, PLUS30
-          move "Assurance", {
-            text "30 damage. As long as the Defending Pokémon’s remaining HP is 60 or less, this attack’s base damage is 60 instead of 30."
-            energyCost F, C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-          move "Hasty Headbutt", {
-            text "100 damage. Rampardos does 20 damage to itself. This attack’s damage isn’t affected by Weakness, Resistance, Poké-Powers, Poké-Bodies, or any other effects on the Defending Pokémon."
-            energyCost F, F, F
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (MysteriousTreasures.RAMPARDOS_33, this);
       case DRIFLOON_6:
-        return basic (this, hp:HP060, type:PSYCHIC, retreatCost:1) {
-          weakness D, PLUS10
-          resistance C, MINUS20
-          move "Blowing Wind", {
-            text "Flip a coin. If heads, put 1 of your Benched Pokémon and all cards attached to it on top of your deck. Shuffle your deck afterward."
-            energyCost P
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-          move "Ominous Wind", {
-            text "10 damage. Flip a coin. If heads, the Defending Pokémon is now Confused and can’t retreat during your opponent’s next turn."
-            energyCost P, C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (DiamondPearl.DRIFLOON_46, this);
       case GIBLE_7:
         return basic (this, hp:HP050, type:COLORLESS, retreatCost:1) {
           weakness C, PLUS10
-          move "", {
-            text "Oran Berry. Remove 1 damage counter from Gible at the end of your turn."
-            energyCost ()
-            attackRequirement {}
-            onAttack {
-              damage 0
+          customAbility {
+            // Oran Berry: Remove 1 damage counter from Gible at the end of your turn.
+            delayedA {
+              before BEGIN_TURN,{
+                if (self.numberOfDamageCounters >= 1) {
```

Comment:
Oran berry should only activate at the end of its owner's turn

Reply:
BETWEEN_TURNS doesn't change anything. It's still happening at the end of both player's turns.